### PR TITLE
move the Docker engine image builds into the os-images repo

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

step 1 of many simplifications :)

@ibuildthecloud @joshwget 

because we're not really versioning anything in the os-engines repo, i recon we can just delete it.

### CODE REVIEW

File: Dockerfile.dapper [hunk 1]
```diff
@@ -53,5 +53,26 @@ RUN curl -fL ${!DOCKER_URL} > /usr/bin/docker && \
...
+RUN curl -sfL https://get.docker.com/builds/Linux/x86_64/docker-1.11.2.tgz | tar xzf - -C /assets && \
+    mv /assets/docker /assets/docker-1.11.2
+RUN curl -sfL https://github.com/rancher/docker/releases/download/v1.11.2-ros1/docker-1.11.2_arm.tgz | tar xzf - -C /assets && \
+    mv /assets/docker /assets/docker-1.11.2_arm
+RUN curl -sfL https://github.com/rancher/docker/releases/download/v1.11.2-ros1/docker-1.11.2_arm64.tgz | tar xzf - -C /assets && \
+    mv /assets/docker /assets/docker-1.11.2_arm64
+RUN curl -sfL https://get.docker.com/builds/Linux/x86_64/docker-1.12.1.tgz | tar xzf - -C /assets && \
+    mv /assets/docker /assets/docker-1.12.1
+
+ENV TEST=1
```
Comment (joshwget) [inline Dockerfile.dapper:75]: Did you mean to leave this in? I'm not sure how this relates.

File: scripts/build-images [hunk 2]
```diff
@@ -12,7 +12,15 @@ mkdir -p dist
...
     name="os-$(echo ${i} | cut -f2 -d-)"
+
+    if [ "$name" == "os-docker" ]; then
+        # VERSION tag for docker engine images
+        IFS='-' read -ra dirname <<< "$i"
+        VERSION=${dirname[2]}
+    fi
+
     tag="${OS_REPO}/${name}:${VERSION}${SUFFIX}"
+    # TODO: try pulling an existing image first
```
Comment (joshwget) [inline scripts/build-images:23]: Not really sure what's meant by this TODO, but I'm assuming you know :smile:

File: scripts/build-images [hunk 3]
```diff
@@ -12,7 +12,15 @@ mkdir -p dist
 
 for i in $BASE/[1-9]*; do
     name="os-$(echo ${i} | cut -f2 -d-)"
+
+    if [ "$name" == "os-docker" ]; then
+        # VERSION tag for docker engine images
+        IFS='-' read -ra dirname <<< "$i"
```
Comment (ibuildthecloud) [inline scripts/build-images:18]: OMG, did you just use IFS, and <<< in the same line.  BASH ACHIEVEMENTS UNLOCKED!

**Review by ibuildthecloud — COMMENTED** [review]
I don't really know if I'm in favor of this.  os-engines was always weird to release.  Because you accept a PR and then build all the images.  The version of the image is the docker version so the question arises after a build do I push all built images or just the new one I care about (like new 1.12.x).  If I push the images I could be mutating a tag.  But what if something changed in the packaging such that the new tag X you build is different from the old tag X.  In short os-engines just seems odd and this PR seems like it's moving the oddity to os-images.   Unless you changed the tagging in some better way.

File: scripts/build-images (hunk 3 above)
Comment (joshwget) [reply scripts/build-images:18]: I think I wrote this. And I have no idea what it does.

File: Dockerfile.dapper (hunk 1 above)
Comment (SvenDowideit) [reply Dockerfile.dapper:75]: yup - thats how the engine images were built - I don't want to change how the code works in the same PR as the move between repos.

File: scripts/build-images (hunk 3 above)
Comment (SvenDowideit) [reply scripts/build-images:18]: no, again, I haven't changed anything :) this is what is in the os-engine repo :)

**Comment by SvenDowideit** [comment]
@ibuildthecloud yes, I don't think we should re-push unless there is a substantive change. I also don't see any reason that we're tagging those images with the OS-release version.

This PR explores the idea that we don't need to build all images with the OS version number - we build when there is an actual change to the image.

I agree that we need to figure out a better tagging - :/

oh, and as an extra fun - I think this PR will be superceeded by one merging os-images into os-services - so everything that defines images are in one repo.

The main driver for what I'm doing is to simplify the end-to-end build script - cos atm, there are way too many repos with way too many snowflakes.

**Comment by SvenDowideit** [comment]
oh. @ibuildthecloud just to make sure we're on the same page - this PR doesn't change the image tags for the docker-engine images - so from my pov, the only thing we need to do in addition to this, is to make the image push conditional - don't push if there have been no changes based on the layer content hash (which might be a useful addition to dapper?)

**Comment by ibuildthecloud** [comment]
I think this just ends up being a discussion on tagging strategy.  No reason to make this innocent bystander PR be responsible for that discussion.

LGTM

File: scripts/build-images (hunk 2 above)
Comment (SvenDowideit) [reply scripts/build-images:23]: most of the time, there's no point re-building - and really, if we do build a new image, we should give it a new tag

**Comment by SvenDowideit** [comment]
I'm going to close this and make the same change in os-services (just tested it, and it works as expected)

**Pull request closed without merging**